File: backend/internal/models/message.go
package models

import "time"

type MessageType string

const (
	SessionCreated MessageType = "session_created"
	Pong MessageType = "pong"
	ColonyState MessageType = "colony_state"
	ColonyUpdated MessageType = "colony_updated"
)

type GameMessage struct {
	Type      MessageType      `json:"type"`
	SessionID string      `json:"session_id,omitempty"`
	Data      interface{} `json:"data,omitempty"`
	Timestamp time.Time   `json:"timestamp"`
}

File: frontend/src/types/message.ts
export enum MessageType {
  SessionCreated = 'session_created',
  Pong = 'pong',
  ColonyState = 'colony_state',
  ColonyUpdated = 'colony_updated',
  GetColony = 'get_colony',
  Directive = 'directive',
}

export interface GameMessage<T> {
  type: MessageType;
  session_id?: string;
  data: T;
  timestamp: string;
}
